pandas/core/groupby/groupby.py

```diff
@@ -1519,7 +1519,10 @@ def _bool_agg(self, val_test, skipna):
 
         def objs_to_bool(vals: ArrayLike) -> tuple[np.ndarray, type]:
             if is_object_dtype(vals):
-                vals = np.array([bool(x) for x in vals])
+                # GH#37501: don't raise on pd.NA when skipna=True
+                vals = np.array(
+                    [bool(x) if not (skipna and isna(x)) else True for x in vals]
+                )
```

Comment:
While slightly more verbose, it seems better to me to avoid evaluating `skipna` at each iteration:

```
if skipna:
    vals = np.array(
        [bool(x) if not isna(x) else True for x in vals]
    )
else:
    vals = np.array([bool(x) for x in vals])
```

Reply:
Sounds good, have changed